tools/flash-shelly.py

```diff
@@ -0,0 +1,664 @@
+    if not info:
+      return None
+    if self.fw_type == 'homekit':
+      version = info['version']
+    elif self.fw_type == 'stock':
+      version = self.parse_stock_version(info['fw'])
+    return version
+
+  def shelly_model(self, type):
+    options = {'SHSW-1' : 'Shelly1',
+               'SHSW-L' : 'Shelly1L',
+               'SHSW-PM' : 'Shelly1PM',
+               'SHSW-21' : 'Shelly2',
+               'SHSW-25' : 'Shelly25',
+               'SHPLG-1' : 'ShellyPlug',
+               'SHPLG2-1' : 'ShellyPlug',
+               'SHPLG-S' : 'ShellyPlugS',
+               'SHPLG-U1' : 'ShellyPlugUS',
+               'SHIX3-1' : 'ShellyI3',
+               'SHBTN-1' : 'ShellyButton1',
+               'SHBLB-1' : 'ShellyBulb',
+               'SHVIN-1' : 'ShellyVintage',
+               'SHBDUO-1' : 'ShellyDuo',
+               'SHDM-1' : 'ShellyDimmer1',
+               'SHDM-2' : 'ShellyDimmer2',
+               'SHRGBW2' : 'ShellyRGBW2',
+               'SHDW-1' : 'ShellyDoorWindow1',
+               'SHDW-2' : 'ShellyDoorWindow2',
+               'SHHT-1' : 'ShellyHT',
+               'SHSM-01' : 'ShellySmoke',
+               'SHWT-1' : 'ShellyFlood',
+               'SHGS-1' : 'ShellyGas',
+               'SHEM' : 'ShellyEM',
+               'SHEM-3' : 'Shelly3EM',
+               'switch1' : 'Shelly1',
+               'switch1pm' : 'Shelly1PM',
+               'switch2' : 'Shelly2',
+               'switch25' : 'Shelly25',
+               'shelly-plug-s' : 'ShellyPlugS',
+               'dimmer1' : 'ShellyDimmer1',
+               'dimmer2' : 'ShellyDimmer2',
+               'rgbw2' : 'ShellyRGBW2',
+    }
+    return options.get(type, type)
+
+  def update_homekit(self, release_info=None):
+    self.flash_fw_type_str = 'HomeKit'
+    self.flash_fw_type = 'homekit'
+    for i in release_info:
+      if self.variant:
+        re_search = '-*'
+      else:
+        re_search = i[0]
+      if re.search(re_search, self.fw_version):
+        self.flash_fw_version = i[1]['version']
+        if not self.version:
+          self.dlurl = i[1]['urls'][self.model] if self.model in i[1]['urls'] else None
+        else:
+          self.dlurl = f'http://rojer.me/files/shelly/{self.version}/shelly-homekit-{self.model}.zip'
+        break
+
+  def update_stock(self, release_info=None):
+    self.flash_fw_type_str = 'Stock'
+    self.flash_fw_type = 'stock'
+    stock_model_info = release_info['data'][self.stock_model]
+    self.flash_fw_version = self.parse_stock_version(stock_model_info['version'])
+    if not self.version:
+      if self.stock_model  == 'SHRGBW2':
+        self.dlurl = stock_model_info['url'].replace('.zip',f'-{self.colour_mode}.zip')
+      else:
+        self.dlurl = stock_model_info['url']
+    else:
+      self.dlurl = f'http://archive.shelly-faq.de/version/v{self.version}/{self.stock_model}.zip'
+
+class HomeKitDevice(Device):
+  def __init__(self, host, wifi_ip, fw_type, device_url, info, variant, version):
```

Comment:
no need to define `__init__` if you're not overriding anything

Reply:
done